pallets/acurast/src/lib.rs

```diff
@@ -315,43 +317,34 @@ pub mod pallet {
             let who = ensure_signed(origin.clone())?;
             let requester = T::Lookup::lookup(requester)?;
 
-            // find assignment
-            let job_id: JobId<T::AccountId> = (requester, fulfillment.script.clone());
-            let mut assigned_jobs = <StoredJobAssignment<T>>::get(&who).unwrap_or_default();
-            let job_index = assigned_jobs
-                .iter()
-                .position(|assigned_job_id| assigned_job_id == &job_id)
-                .ok_or(Error::<T>::FulfillSourceNotAllowed)?;
-
             // find registration
-            let registration = <StoredJobRegistration<T>>::get(&job_id.0, &fulfillment.script)
-                .ok_or(Error::<T>::JobRegistrationNotFound)?;
+            let registration =
+                <StoredJobRegistration<T>>::get(&requester.clone(), &fulfillment.script)
+                    .ok_or(Error::<T>::JobRegistrationNotFound)?;
 
             ensure_source_allowed::<T>(&who, &registration)?;
 
-            T::RewardManager::pay_reward(
-                registration.reward.clone(),
-                T::Lookup::unlookup(who.clone()),
+            <T as Config>::JobHooks::fulfill_hook(
+                &who,
+                &fulfillment,
+                requester.clone(),
+                &registration,
             )?;
 
-            // route fulfillment
+            // only route fulfillment after all checks succeeded since this call is not revertable
```

Comment:
I am 99% sure this is not true.

Everything should get reverted, if the call returns `Err`.

Reply:
removed comment